diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index b82a593..cf72069 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -811,7 +811,7 @@ class Page(paginate.Page):
     def pager(self, *args, **kwargs):
         kwargs.update(
             format=u"<div class='pagination pagination-centered'><ul>$link_previous ~2~ $link_next</ul></div>",
-            symbol_previous=u'«', symbol_next=u'»',
+            symbol_previous=u'Previous Page', symbol_next=u'Next Page',
             curpage_attr={'class': 'active'}, link_attr={}
         )
         return super(Page, self).pager(*args, **kwargs)
@@ -831,8 +831,8 @@ class Page(paginate.Page):
         html = super(Page, self)._range(regexp_match)
         # Convert ..
         dotdot = '<span class="pager_dotdot">..</span>'
-        dotdot_link = HTML.li(HTML.a('...', href='#'), class_='disabled')
-        html = re.sub(dotdot, dotdot_link, html)
+
+        html = re.sub(dotdot, '', html)
 
         # Convert current page
         text = '%s' % self.page
