diff --git a/ckanext/reclineview/theme/public/vendor/recline/recline.js b/ckanext/reclineview/theme/public/vendor/recline/recline.js
index 42be873..37f1cf8 100755
--- a/ckanext/reclineview/theme/public/vendor/recline/recline.js
+++ b/ckanext/reclineview/theme/public/vendor/recline/recline.js
@@ -2283,8 +2283,8 @@ my.Map = Backbone.View.extend({
     var self = this;
     this.map = new L.Map(this.$map.get(0));
 
-    var mapUrl = "//otile{s}-s.mqcdn.com/tiles/1.0.0/osm/{z}/{x}/{y}.png";
-    var osmAttribution = 'Map data &copy; 2011 OpenStreetMap contributors, Tiles Courtesy of <a href="http://www.mapquest.com/" target="_blank">MapQuest</a> <img src="//developer.mapquest.com/content/osm/mq_logo.png">';
+    var mapUrl = "//c.tiles.mapbox.com/v3/maxious.gdb67hbh/{z}/{x}/{y}.png";
+    var osmAttribution = "Map data &copy; OpenStreetMap contributors, <a href='https://www.mapbox.com/about/maps/' target='_blank'>&copy; Mapbox</a> <a class='mapbox-improve-map' href='https://www.mapbox.com/map-feedback/' target='_blank'>Improve this map</a> ";
     var bg = new L.TileLayer(mapUrl, {maxZoom: 18, attribution: osmAttribution ,subdomains: '1234'});
     this.map.addLayer(bg);
 
@@ -3222,7 +3222,12 @@ my.SlickGrid = Backbone.View.extend({
     }
 
     function sanitizeFieldName(name) {
-      var sanitized = $(name).text();
+ var sanitized;
+      try{
+        sanitized = $(name).text();
+      } catch(e){
+        sanitized = '';
+      }
       return (name !== sanitized && sanitized !== '') ? sanitized : name;
     }
 
