diff --git a/ckan/controllers/package.py b/ckan/controllers/package.py
index e3aa7c8..7a9d5ed 100644
--- a/ckan/controllers/package.py
+++ b/ckan/controllers/package.py
@@ -546,7 +546,10 @@ class PackageController(base.BaseController):
         error_summary = error_summary or {}
         # in the phased add dataset we need to know that
         # we have already completed stage 1
-        stage = ['active']
+        if 'unpublished' in data and data['unpublished'] == 'True':
+            stage = ['active', 'complete']
+        else:
+            stage = ['active']
         if data.get('state', '').startswith('draft'):
             stage = ['active', 'complete']
 
@@ -946,8 +949,9 @@ class PackageController(base.BaseController):
                                         id=pkg_dict['name'])
                     redirect(url)
                 # Make sure we don't index this dataset
-                if request.params['save'] not in ['go-resource', 'go-metadata']:
-                    data_dict['state'] = 'draft'
+                if 'unpublished' not in data_dict or data_dict['unpublished'] != 'True':
+                    if request.params['save'] not in ['go-resource', 'go-metadata']:
+                        data_dict['state'] = 'draft'
                 # allow the state to be changed
                 context['allow_state_change'] = True
 
@@ -956,10 +960,16 @@ class PackageController(base.BaseController):
             pkg_dict = get_action('package_create')(context, data_dict)
 
             if ckan_phase:
-                # redirect to add dataset resources
-                url = h.url_for(controller='package',
-                                action='new_resource',
-                                id=pkg_dict['name'])
+                if 'unpublished' in pkg_dict and pkg_dict['unpublished'] == 'True':
+                    # redirect to add dataset resources
+                    url = h.url_for(controller='package',
+                                    action='read',
+                                    id=pkg_dict['name'])
+                else:
+                    # redirect to add dataset resources
+                    url = h.url_for(controller='package',
+                                    action='new_resource',
+                                    id=pkg_dict['name'])
                 redirect(url)
 
             self._form_save_redirect(pkg_dict['name'], 'new', package_type=package_type)
